internal/publish/serverservice_components.go

```diff
@@ -881,3 +911,42 @@ func (h *serverServicePublisher) setVersionedAttributes(component *serverservice
 		},
 	)
 }
+
+// addFirmwareData queries ServerService for the firmware version and try to find a match.
+func (h *serverServicePublisher) addFirmwareData(ctx context.Context, deviceVendor string, component *serverservice.ServerComponent, vattr *versionedAttributes) (vatrr *versionedAttributes, err error) {
+	// Filter by firmware vendor + version
+	// TODO: fix serverservice to accept component in filter
```

Comment:
If it helps, its possible to filter serverservice components by type with - `/api/v1/servers/components?sc_0[type]=drive`

Reply:
This is the component on the `component_firmware_version` table. At this point in the code we already have the component we are dealing with in the ServerComponent object.